src/py42/services/users.py

```diff
@@ -380,3 +393,29 @@ def update_user(
             elif "INVALID_USERNAME" in response_text:
                 raise Py42InvalidUsernameError(err)
             raise
+
+    def _get_user_uid_by_id(self, user_id):
+        # Identity crisis helper method.
+        # Old py42 methods accepted IDs. New apis take UIDs.
+        # Use additional lookup to prevent breaking changes.
+        return self.get_by_id(user_id)["userUid"]
+
+    def _get_role_ids(self, user_id):
+        return [i["roleId"] for i in self.get_roles(user_id)]
+
+    def _update_role_ids(self, role_name, role_ids, add=True):
+
+        for role in self.get_available_roles():
+            if role["roleName"] == role_name:
+                if add:
+                    role_ids.append(role["roleId"])
+                else:
+                    role_ids.remove(role["roleId"])
+                break
+
+        return role_ids
+
+    def _update_roles(self, user_id, role_ids):
```

Comment:
Since we're already doing the lookup either way, we could just accept role-id _or_ role-name values and use them if they're valid, just make the check `if (role["roleName"] == role_name) or (role["roleId"] == role_name):`

Then just update the docs like:

```
role_name (str): The name or id of the role to assign to the user. e.g. "Desktop User" (name) or "desktop-user" (id)
```

This keeps it backwards compatible, but makes it easy to re-use the role-id values. I was testing these changes in the CLI and noticed the CLI only returns `role-id` in user results, but requires `role-name` in the arguments, making this change would make that awkward inconsistency go away.